src/platform/forms-system/src/js/components/FormNav.jsx

```diff
@@ -10,72 +9,96 @@ import {
+import { focusElement } from '../utilities/ui';
+
 import PropTypes from 'prop-types';
 import { REVIEW_APP_DEFAULT_MESSAGE } from '../constants';
 
-export default class FormNav extends React.Component {
-  // The formConfig transforming is a little heavy, so skip it if we can
-  shouldComponentUpdate(newProps) {
-    return !shallowEqual(this.props, newProps);
-  }
-  render() {
-    const { formConfig, currentPath, formData } = this.props;
+export default function FormNav(props) {
+  const { formConfig, currentPath, formData } = props;
+
+  const [index, setIndex] = useState(0);
+
+  // This is converting the config into a list of pages with chapter keys,
+  // finding the current page, then getting the chapter name using the key
+  const formPages = createFormPageList(formConfig);
+  const pageList = createPageList(formConfig, formPages);
 
-    // This is converting the config into a list of pages with chapter keys,
-    // finding the current page, then getting the chapter name using the key
-    const formPages = createFormPageList(formConfig);
-    const pageList = createPageList(formConfig, formPages);
+  const eligiblePageList = getActiveExpandedPages(pageList, formData);
 
-    const eligiblePageList = getActiveExpandedPages(pageList, formData);
+  const chapters = _.uniq(
+    eligiblePageList.map(p => p.chapterKey).filter(key => !!key),
+  );
 
-    const chapters = _.uniq(
-      eligiblePageList.map(p => p.chapterKey).filter(key => !!key),
+  let page = eligiblePageList.filter(p => p.path === currentPath)[0];
+  // If the page isn’t active, it won’t be in the eligiblePageList
+  // This is a fallback to still find the chapter name if you open the page directly
+  // (the chapter index will probably be wrong, but this isn’t a scenario that happens in normal use)
+  if (!page) {
+    page = formPages.find(
+      p => `${formConfig.urlPrefix}${p.path}` === currentPath,
     );
+  }
 
-    let page = eligiblePageList.filter(p => p.path === currentPath)[0];
-    // If the page isn’t active, it won’t be in the eligiblePageList
-    // This is a fallback to still find the chapter name if you open the page directly
-    // (the chapter index will probably be wrong, but this isn’t a scenario that happens in normal use)
-    if (!page) {
-      page = formPages.find(
-        p => `${formConfig.urlPrefix}${p.path}` === currentPath,
-      );
+  let current;
+  let chapterName;
+  if (page) {
+    current = chapters.indexOf(page.chapterKey) + 1;
+    // The review page is always part of our forms, but isn’t listed in chapter list
+    chapterName =
+      page.chapterKey === 'review'
+        ? formConfig?.customText?.reviewPageTitle || REVIEW_APP_DEFAULT_MESSAGE
+        : formConfig.chapters[page.chapterKey].title;
+    if (typeof chapterName === 'function') {
+      chapterName = chapterName();
     }
+  }
+
+  const stepText = `Step ${current} of ${chapters.length}: ${chapterName}`;
+  const showHeader = Math.abs(current - index) === 1;
```

Comment:
We want to render the `<h2>` header anytime the `current` value is only one integer ahead or behind the `index`.